src/actions/accounts.js

```diff
@@ -96,7 +96,6 @@ export const onAccountDelete = async (client, account) => {
 
 export const onGroupDelete = async (client, account) => {
   await deleteOrphanOperations(client, account)
-  await removeAccountFromGroups(client, account)
```

Comment:
Why not having fixed the removeAccountFromGroups? The race condition between action doesn't small good. We should not have this kind of behavior. 

After a quick check it seems that `removeAccountFromGroups` bypass CozyClient links by calling directly the stack client : 
```
const groupCollection = client.stackClient.collection(GROUP_DOCTYPE)
  const groups = (await groupCollection.all()).data
  const ugroups = groups.map(group => removeAccountFromGroup(group, account))
  for (let ugroup of ugroups) {
    await groupCollection.update(ugroup)
  }
```

If we change that by: 
```
  const groups = (await client.query(Q(GROUP_DOCTYPE)).data
  const ugroups = groups.map(group => removeAccountFromGroup(group, account))
await client.saveAll(ugroups)
```

Do we still have the issue?

Reply:
Also, when this `removeAccountFromGroups` is done now with this fix? Where do we remove the account from the group?

Reply:
Je vais essayer via client.query.
Pour la deuxième question, on ne le fait pas du coup. Mais c'est pas bien grave puisqu'on supprime le groupe, non ?

Reply:
Oui mais on parle de `removeAccountFromGroups` au pluriel, pas juste le groupe courant. De ce que je comprends du code, quand on supprime un groupe on récupère un account, on boucle sur tous les groupes existants et on enlève la liaison avec cet account. 

Mais je suis trop loin pour savoir ce qu'est cet account et si ça fait sens de boucler sur tous les groupes pour le retirer

Reply:
Non a priori ici ça n'a pas de sens, il n'y a pas de manipulation sur l'account (c'est le compte bancaire). Donc pas trop besoin de boucler sur les groups pour faire quoi que ce soit en rapport avec l'account.

Historiquement on avait que la suppression de compte, processus dans lequel on devait faire un certain nombre de chose, dont la suppression de l'account dans les groupes. Puis via https://github.com/cozy/cozy-banks/pull/1996/files on a introduit le fait de désactiver les notifs, et la suppression d'un groupe (voir mon commit https://github.com/cozy/cozy-banks/pull/2229/commits/876996ec15dcbfd5cb6e894c9e34e5e1a5004a55). Et à mon avis à ce moment là on a fait un copier-coller sans trop se poser de question.

Reply:
OK ça marche merci.

Ne pas oublier de corriger les appels à collection() quand on les voit passer sous nos yeux ;)

Reply:
Oui bien vu pour ça, je me garde ça sous le coude pour tester ce que tu indiques dans le premier commentaire.